### PR TITLE
UIIN-1374: Add ability to mark item as Intellectual item and Restricted

### PR DESCRIPTION
**Pull request by SergiyVSergiyenko**
https://issues.folio.org/browse/UIIN-1374

### Purpose
It was added the ability to mark an item as intellectual and restricted from the item view page.
The code for displaying the modal and the tests for checking it were worked out so that the rest of the item statuses from the extended list ([UIIN-756](https://issues.folio.org/browse/UIIN-756)) can be added smoothly.

### Screenshot
Screen Shot 2021-01-15 at 19 35 05

### CODE REVIEW

**Review by Baroquem — APPROVED** [review]
Nice work!

File: src/constants.js [hunk 1]
```diff
@@ -18,6 +18,15 @@ export const itemStatusesMap = {
...
+  ORDER_CLOSED: 'Order closed',
+  INTELLECTUAL_ITEM: 'Intellectual item',
+  RESTRICTED: 'Restricted',
+};
+
+// Matching mutator names to the corresponding item status
+export const itemStatusMutators = {
+  INTELLECTUAL_ITEM: 'markAsIntellectualItem',
+  RESTRICTED: 'markAsRestricted',
 };
```
Comment (zburke) [inline src/constants.js:30]: Clever; nice!

**Review by zburke — APPROVED** [review]
LGTM!

**Review by mkuklis — APPROVED** [review]
Nice and clean :)

**Review by ViktorSoroka07 — APPROVED** [review]

**Pull request merged**